pkg/internal/appolly/appolly.go

```diff
@@ -131,14 +133,16 @@ func (i *Instrumenter) FindAndInstrument(ctx context.Context) error {
 					}
 					i.processEventInput.Send(exec.ProcessEvent{Type: exec.ProcessEventCreated, File: pt.FileInfo})
 				case discover.EventDeleted:
-					dp := ev.Obj
 					log.Debug("stopping ProcessTracer because there are no more instances of such process",
-						"inode", dp.FileInfo.Ino, "pid", dp.FileInfo.Pid, "exec", dp.FileInfo.CmdExePath)
-					if dp.Tracer != nil {
-						dp.Tracer.UnlinkExecutable(dp.FileInfo)
+						"inode", pt.FileInfo.Ino, "pid", pt.FileInfo.Pid, "exec", pt.FileInfo.CmdExePath)
+					if pt.Tracer != nil {
+						pt.Tracer.UnlinkExecutable(pt.FileInfo)
 					}
-					i.processEventInput.Send(exec.ProcessEvent{Type: exec.ProcessEventTerminated, File: dp.FileInfo})
+					i.processEventInput.Send(exec.ProcessEvent{Type: exec.ProcessEventTerminated, File: pt.FileInfo})
 				case discover.EventInstanceDeleted:
+					if pt.Tracer != nil {
```

Comment:
Do we really need this here? The difference between EventDeleted and EventInstanceDeleted is that with the "instance" flavour of the event we know we still have tracers live.